tests/pgautofailover_utils.py

```diff
@@ -331,29 +331,42 @@ def wait_until_state(self, target_state, timeout=STATE_CHANGE_TIMEOUT):
         prev_state = None
         for i in range(timeout):
             # ensure we read from the pg_autoctl process pipe
-            self.pg_autoctl.consume_output(1)
+            if self.pg_autoctl and self.pg_autoctl.run_proc:
+                self.pg_autoctl.consume_output(1)
+            else:
+                time.sleep(1)
 
             current_state = self.get_state()
 
             # only log the state if it has changed
             if current_state != prev_state:
-                print("state of %s is '%s', waiting for '%s' ..." %
-                    (self.datadir, current_state, target_state))
+                if current_state == target_state:
```

Comment:
we have the same check twice

can we have something like
```python
if current_state == target_state:
    print("state of %s is '%s', waiting for '%s' ..." %(self.datadir, current_state, target_state))
    return True
```

above the enclosing if ?

Reply:
I think that if `current_state == target_state` then we are done, and we want to print something only when previous state was either None or different from the current_state. In other words I'm not sure we can simplify the logic down much.